tests/mu_define_enum_test.c

```diff
@@ -34,5 +66,62 @@ int run_mu_define_enum_tests(void)
 
     POOR_MANS_ASSERT(MU_ENUM_VALUE_COUNT(TEST4_ENUM_VALUES) == 3);
 
+    const char* TEST5_test5_b = MU_ENUM_TO_STRING(TEST5_ENUM, 42);
+    POOR_MANS_ASSERT(TEST5_test5_b != NULL);
+    POOR_MANS_ASSERT(strcmp("test5_b = 42", TEST5_test5_b) == 0);
+
+    const char* TEST5_unknown = MU_ENUM_TO_STRING(TEST5_ENUM, 41);
+    POOR_MANS_ASSERT(TEST5_unknown != NULL);
+    POOR_MANS_ASSERT(strcmp("UNKNOWN", TEST5_unknown) == 0);
+
+    const char* TEST6_test6_a = MU_ENUM_TO_STRING(TEST6_ENUM, test6_a);
+    POOR_MANS_ASSERT(TEST6_test6_a != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_a = 42", TEST6_test6_a) == 0);
+
+    // test6_b and TEST6_ENUM_INVALID are both 0
+    const char* TEST6_test6_b = MU_ENUM_TO_STRING(TEST6_ENUM, test6_b);
```

Comment:
>const char* TEST6_test6_b = MU_ENUM_TO_STRING(TEST6_ENUM, test6_b); [](http://example.com/codeflow?start=4&length=67)

How does this work for SF enums?

I would suppose those enums (that we print) do have "0" as a valid value.

Reply:
Well the code gets the string for the first 0 value in the enum
I believe the SF enums we have are without invalid.

---
In reply to: [632129694](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632129694) [](ancestors = 632129694)